Dockerfile

```diff
@@ -1,31 +1,59 @@
-FROM python:3.8-slim
+# ---
+# --- Builder image
+# ---
 
-RUN apt-get update && apt-get -y install git gcc
+FROM python:3.10-alpine as builder
 
-RUN useradd -r app_user
-RUN mkdir /app
-RUN chown app_user /app/
-WORKDIR /app
+# Install build dependancies
+RUN apk update && apk add --upgrade --no-cache g++ make
 
-# Installing packages
-RUN pip install pipenv==2018.11.26
+# Install pipenv
+RUN pip install --no-cache-dir --upgrade pip
+RUN pip install pipenv
 
-ADD --chown=app_user app.py Pipfile* /app/
-RUN pipenv install --system --deploy
+WORKDIR /usr/local/src
 
-USER app_user
+# Build a venv with dependancies in current directory
+ADD Pipfile.lock Pipfile* /usr/local/src/
+RUN PIPENV_VENV_IN_PROJECT=1 pipenv sync
 
-# the sources are copied as late as possible since they are likely to change often
-ADD --chown=app_user idunn /app/idunn
+# ---
+# --- Application image
+# ---
 
-# set the multiprocess mode for gunicorn
+FROM python:3.10-alpine as runtime
+
+# Create the user idunn
+RUN addgroup --gid 1000 idunn
+RUN adduser --disabled-password --home /home/idunn --ingroup idunn \
+            --uid 1000 idunn
+
+ENV PYTHONUNBUFFERED=1
+
+# Set the multiprocess mode for gunicorn
 ENV IDUNN_PROMETHEUS_MULTIPROC=1
-ENV PROMETHEUS_MULTIPROC_DIR=/app/idunn/prometheus_multiproc
-RUN mkdir /app/idunn/prometheus_multiproc
+ENV PROMETHEUS_MULTIPROC_DIR=/home/idunn/prometheus_multiproc
+
+# Install lib dependancies
+RUN apk update && apk add --upgrade --no-cache geos
+
+RUN mkdir -p /home/idunn/prometheus_multiproc
+USER idunn
+WORKDIR /home/idunn
+
+# Add files into images
+ADD app.py /home/idunn
+ADD idunn /home/idunn/idunn
+COPY --chown=idunn --from=builder /usr/local/src/.venv /home/idunn/.venv
```

Comment:
Are you sure this change of ownership is required? Better to leave the binary owned by root (not writeable by the used that will execute the application).